Mimir.Worker/Handler/LegacyAccountHandler.cs

```diff
@@ -45,12 +53,86 @@ private State ConvertToState(Address address, IValue rawState)
                     );
                 }
 
-                return ConvertToTableSheetState(sheetType, addr, rawState);
+                return ConvertToTableSheetState(sheetType, addr, context.RawState);
+            case var addr
+                when derivedAddresses.TryGetValue(
+                    "ArenaScoreAddresses",
+                    out var arenaScoreAddresses
+                ) && arenaScoreAddresses.Contains(addr):
+                return ConvertToArenaScoreState(addr, context.RawState);
+            case var addr
+                when derivedAddresses.TryGetValue("ArenaInfoAddresses", out var arenaInfoAddresses)
+                    && arenaInfoAddresses.Contains(addr):
+                return ConvertToArenaInformationState(addr, context.RawState);
             default:
                 throw new InvalidOperationException("The provided address has not been handled.");
         }
     }
 
+    private Dictionary<string, List<Address>> GetDerivedFromAvatarsAddresses(List<Transaction> txs)
+    {
+        var addresses = new Dictionary<string, List<Address>>
+        {
+            { "ArenaScoreAddresses", new List<Address>() },
+            { "ArenaInfoAddresses", new List<Address>() }
+        };
+
+        foreach (var tx in txs)
+        {
+            var action = (Dictionary)tx.Actions[0];
+            var actionType = (Text)action["type_id"];
+            var actionValues = action["values"];
+
+            if (Regex.IsMatch(actionType, "^battle_arena[0-9]*$"))
+            {
+                var avatarAddress = new Address(((Dictionary)actionValues)["maa"]);
```

Comment:
Isn't okay to check `eaa`, `EnemyAvatarAddress` too? It can be invalid memory, I remember the enemy lost its arena point if enemy fails.

Reply:
Score changes in BattleArena action: https://github.com/planetarium/lib9c/blob/development/Lib9c/Action/BattleArena.cs#L450-L464

Reply:
Oh, my bad i'll check enemy avatar too

Reply:
Done!